tsdb/series_index.go

```diff
@@ -155,6 +155,14 @@ func (idx *SeriesIndex) Recover(segments []*SeriesSegment) error {
 	return nil
 }
 
+// GrowBy preallocates the in-memory hashmap to a larger size.
+func (idx *SeriesIndex) GrowBy(delta int) {
+	if delta < 0 {
+		return
+	}
+	idx.keyIDMap.Grow(((idx.keyIDMap.Len() + int64(delta)) * 100) / int64(idx.keyIDMap.LoadFactor()))
```

Comment:
can `LoadFactor` ever be `0`?

Reply:
Under what conditions would a `delta` value result in less than a doubling of the map? Seems to me at a minimum we always want to at least double the map right? So maybe we should set a min value for `delta`?

Reply:
`LoadFactor` of `0` would have serious issues way before it got to the point of needing to grow. We only adjust the load factor via a constant right now anyway.

re: `delta`, the `rhh.Map` requires that the capacity be a power of two so it will automatically do the doubling.